packages/amplify-provider-awscloudformation/lib/build-resources.js

```diff
@@ -14,134 +11,90 @@ async function run(context, category, resourceName) {
-  const lastUpdateTime = fs.readdirSync(resourceDir)
-    .map(file => path.join(resourceDir, file))
-    .map(file => fs.statSync(file))
-    .map(stats => stats.mtime)
-    .reduce((acc, it) => it > acc ? it : acc, 0); // reduce to the latest time
-  const distDir = path.normalize(path.join(backEndDir, category, resourceName, 'dist'));
-
-  let zipFilename = resource.distZipFilename;
-  let zipFilePath = zipFilename ? path.normalize(path.join(distDir, 'latest-build.zip')) : '';
 
-  if (!resource.lastBuildTimeStamp || new Date(lastUpdateTime) > new Date(resource.lastBuildTimeStamp)) {
-    installDependencies(resourceDir);
-    context.amplify.updateamplifyMetaAfterBuild(resource);
+// This function is a translation layer around the previous buildResource
+// For legacy purposes, the method builds and packages the resource
+async function buildResource(context, resource) {
+  const resourcePath = path.join(context.amplify.pathManager.getBackendDirPath(), resource.category, resource.resourceName);
+  const breadcrumbsPath = path.join(resourcePath, 'amplify.function');
+  let breadcrumbs = context.amplify.readJsonFile(breadcrumbsPath);
+  if (!breadcrumbs) {
+    // fallback to old behavior for backwards compatibility
+    breadcrumbs = {
+      pluginId: 'amplify-nodejs-function-runtime-provider',
+      functionRuntime: 'nodejs',
+      useLegacyBuild: true,
+    };
   }
 
-  runBuildScriptHook(resourceName, projectRoot);
-
-  if (!resource.lastPackageTimeStamp || !resource.distZipFilename || isPackageOutdated(resourceDir, resource.lastPackageTimeStamp)) {
-    // generating hash, ignoring node_modules as this can take long time to hash
-    // the content inside node_modules change only when content of package-lock.json changes
-    const { hash: folderHash } = await hashElement(resourceDir, {
-      folders: { exclude: ['node_modules'] },
-    });
-
-    zipFilename = `${resourceName}-${Buffer.from(folderHash)
-      .toString('hex')
-      .substr(0, 20)}-build.zip`;
-
-    if (!fs.existsSync(distDir)) {
-      fs.mkdirSync(distDir);
-    }
-
-    zipFilePath = path.normalize(path.join(distDir, 'latest-build.zip'));
-    const output = fs.createWriteStream(zipFilePath);
+  let zipFilename = resource.distZipFilename;
 
-    return new Promise((resolve, reject) => {
-      output.on('close', () => {
-        context.amplify.updateAmplifyMetaAfterPackage(resource, zipFilename);
-        resolve({ zipFilePath, zipFilename });
-      });
-      output.on('error', () => {
-        reject(new Error('Failed to zip code.'));
-      });
-      const zip = archiver.create('zip', {});
-      zip.pipe(output);
-      zip.directory(resourceDir, false);
-      zip.finalize();
-    });
+  const runtimePlugin = await loadRuntimePlugin(context, breadcrumbs.pluginId);
+
+  // build the function
+  let rebuilt = false;
+  if (breadcrumbs.scripts && breadcrumbs.scripts.build) {
+    // TODO
```

Comment:
when is it coming, separate PR?

Reply:
It could be this PR, but I'm trying to get all of the new bits of the runtime plugin interface wired up asap so that others can test their plugins e2e and this isn't necessary for that to work